org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeBaseChannelConverterProvider.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.zigbee.internal.converter;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.openhab.binding.zigbee.ZigBeeBindingConstants;
+import org.openhab.binding.zigbee.converter.ZigBeeBaseChannelConverter;
+import org.openhab.binding.zigbee.converter.ZigBeeChannelConverterProvider;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * The base {@link ZigBeeBaseChannelConverterProvider} of the binding making the standard
+ * {@link ZigBeeBaseChannelConverter}s available for the {@link ZigBeeChannelConverterFactory}.
+ *
+ * @author Thomas Höfer - Initial contribution
+ */
+@Component(immediate = true, service = ZigBeeChannelConverterProvider.class)
+public final class ZigBeeBaseChannelConverterProvider implements ZigBeeChannelConverterProvider {
```

Comment:
The method `ZigBeeChannelConverterProvider#getChannelConverters(...)` returns instances of `ZigBeeBaseChannelConverter`. In this light, I find it somewhat misleading that this provider here is called `ZigBeeBasedChannelConverterProvider` - as all such providers provide `ZigBeeBaseChannelConverter`s.

Suggestion: Extract an interface `ZigBeeChannelConverter`, make the `ZigBeeBaseChannelConverterProvider` return instances of this interface, and make the `ZigBeeBaseChannelConverter` class implement this interface.

WDYT?

Reply:
I understand your point. When I started to implement this enhancement I thought about returning an interface. However I don´t know at the moment which operations should be available in this interface. I propose to wait until we have implemented a second provider and get more familiar with converters in general. Then we can still refactor this.

I agree that the term ```ZigBeeBaseChannelConverterProvider``` is not perfect. I will rename it to ```ZigBeeDefaultChannelConverterProvider```